packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -1290,3 +1306,20 @@ Axis? _getPanAxis(Offset point1, Offset point2) {
   final double y = point2.dy - point1.dy;
   return x.abs() > y.abs() ? Axis.horizontal : Axis.vertical;
 }
+
+/// This enum is used to specify the behavior of the [InteractiveViewer] when
+/// the user drags the viewport.
+enum PanAxis{
+  /// The user can pan the viewport along the horizontal axis.
+  horizontal,
+
+  /// The user can pan the viewport along the vertical axis.
+  vertical,
+
+  /// The user can pan the viewport along the horizontal and vertical axes but
+  /// not diagonally.
+  aligned,
+
+  /// The user can pan the viewport along both the horizontal and vertical axes.
+  both,
```

Comment:
Nit: I feel like there's a better word for this instead of "both"... Maybe "free"?

Reply:
Perfect, both sounds kind of wacky, let's use free!